// @shopify/polaris/build/cjs/components/Inline/Inline.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var components = require('../../utilities/components.js');
var Inline$1 = require('./Inline.scss.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const AlignY = {
  top: 'start',
  center: 'center',
  bottom: 'end',
  baseline: 'baseline'
};
const Inline = function Inline({
  children,
  spacing = '1',
  align,
  alignY,
  wrap
}) {
  const style = {
    '--pc-inline-align': align,
    '--pc-inline-align-y': alignY ? AlignY[alignY] : undefined,
    '--pc-inline-wrap': wrap ? 'wrap' : 'nowrap',
    '--pc-inline-spacing': `var(--p-space-${spacing})`
  };
  const itemMarkup = components.elementChildren(children).map((child, index) => {
    return /*#__PURE__*/React__default["default"].createElement("div", {
      key: index
    }, child);
  });
  return /*#__PURE__*/React__default["default"].createElement("div", {
    className: Inline$1["default"].Inline,
    style: style
  }, itemMarkup);
};

exports.Inline = Inline;
