// @shopify/polaris/build/cjs/components/Bleed/Bleed.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var css = require('../../utilities/css.js');
var Bleed$1 = require('./Bleed.scss.js');

function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

var React__default = /*#__PURE__*/_interopDefaultLegacy(React);

const Bleed = ({
  spacing,
  horizontal,
  vertical,
  top,
  bottom,
  left,
  right,
  children
}) => {
  const getNegativeMargins = direction => {
    const xAxis = ['left', 'right'];
    const yAxis = ['top', 'bottom'];
    const directionValues = {
      top,
      bottom,
      left,
      right,
      horizontal,
      vertical
    };

    if (directionValues[direction]) {
      return directionValues[direction];
    } else if (!yAxis.includes(direction) && horizontal) {
      return directionValues.horizontal;
    } else if (!xAxis.includes(direction) && vertical) {
      return directionValues.vertical;
    } else {
      return spacing;
    }
  };

  const negativeMargins = {
    top: getNegativeMargins('top'),
    left: getNegativeMargins('left'),
    right: getNegativeMargins('right'),
    bottom: getNegativeMargins('bottom')
  };
  const style = { ...(negativeMargins.bottom ? {
      '--pc-bleed-margin-bottom': `var(--p-space-${negativeMargins.bottom})`
    } : undefined),
    ...(negativeMargins.left ? {
      '--pc-bleed-margin-left': `var(--p-space-${negativeMargins.left})`
    } : undefined),
    ...(negativeMargins.right ? {
      '--pc-bleed-margin-right': `var(--p-space-${negativeMargins.right})`
    } : undefined),
    ...(negativeMargins.top ? {
      '--pc-bleed-margin-top': `var(--p-space-${negativeMargins.top})`
    } : undefined)
  };
  return /*#__PURE__*/React__default["default"].createElement("div", {
    className: Bleed$1["default"].Bleed,
    style: css.sanitizeCustomProperties(style)
  }, children);
};

exports.Bleed = Bleed;
