// @shopify/polaris/build/esm/components/Bleed/Bleed.js
import React from 'react';
import { sanitizeCustomProperties } from '../../utilities/css.js';
import styles from './Bleed.scss.js';

const Bleed = ({
  spacing,
  horizontal,
  vertical,
  top,
  bottom,
  left,
  right,
  children
}) => {
  const getNegativeMargins = direction => {
    const xAxis = ['left', 'right'];
    const yAxis = ['top', 'bottom'];
    const directionValues = {
      top,
      bottom,
      left,
      right,
      horizontal,
      vertical
    };

    if (directionValues[direction]) {
      return directionValues[direction];
    } else if (!yAxis.includes(direction) && horizontal) {
      return directionValues.horizontal;
    } else if (!xAxis.includes(direction) && vertical) {
      return directionValues.vertical;
    } else {
      return spacing;
    }
  };

  const negativeMargins = {
    top: getNegativeMargins('top'),
    left: getNegativeMargins('left'),
    right: getNegativeMargins('right'),
    bottom: getNegativeMargins('bottom')
  };
  const style = { ...(negativeMargins.bottom ? {
      '--pc-bleed-margin-bottom': `var(--p-space-${negativeMargins.bottom})`
    } : undefined),
    ...(negativeMargins.left ? {
      '--pc-bleed-margin-left': `var(--p-space-${negativeMargins.left})`
    } : undefined),
    ...(negativeMargins.right ? {
      '--pc-bleed-margin-right': `var(--p-space-${negativeMargins.right})`
    } : undefined),
    ...(negativeMargins.top ? {
      '--pc-bleed-margin-top': `var(--p-space-${negativeMargins.top})`
    } : undefined)
  };
  return /*#__PURE__*/React.createElement("div", {
    className: styles.Bleed,
    style: sanitizeCustomProperties(style)
  }, children);
};

export { Bleed };
